src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CachedSystemStoreProvider.cs

```diff
@@ -129,53 +129,50 @@ public void Remove(ICertificatePal cert)
                 Monitor.IsEntered(s_recheckStopwatch),
                 "LoadMachineStores assumes a lock(s_recheckStopwatch)");
 
-            IEnumerable<FileInfo> trustedCertFiles;
+            SafeX509StackHandle rootStore = Interop.Crypto.NewX509Stack();
+            Interop.Crypto.CheckValidOpenSslHandle(rootStore);
+            SafeX509StackHandle intermedStore = Interop.Crypto.NewX509Stack();
+            Interop.Crypto.CheckValidOpenSslHandle(intermedStore);
+
             DateTime newFileTime = default;
             DateTime newDirTime = default;
 
-            if (rootStorePath != null && rootStorePath.Exists)
-            {
-                trustedCertFiles = rootStorePath.EnumerateFiles();
-                newDirTime = rootStorePath.LastWriteTimeUtc;
-            }
-            else
-            {
-                trustedCertFiles = Array.Empty<FileInfo>();
-            }
+            var uniqueRootCerts = new HashSet<X509Certificate2>();
+            var uniqueIntermediateCerts = new HashSet<X509Certificate2>();
 
             if (rootStoreFile != null && rootStoreFile.Exists)
             {
-                trustedCertFiles = trustedCertFiles.Prepend(rootStoreFile);
                 newFileTime = rootStoreFile.LastWriteTimeUtc;
+                ProcessFile(rootStoreFile);
             }
 
-            SafeX509StackHandle rootStore = Interop.Crypto.NewX509Stack();
-            Interop.Crypto.CheckValidOpenSslHandle(rootStore);
-            SafeX509StackHandle intermedStore = Interop.Crypto.NewX509Stack();
-            Interop.Crypto.CheckValidOpenSslHandle(intermedStore);
-
-            HashSet<X509Certificate2> uniqueRootCerts = new HashSet<X509Certificate2>();
-            HashSet<X509Certificate2> uniqueIntermediateCerts = new HashSet<X509Certificate2>();
+            if (rootStorePath != null && rootStorePath.Exists)
```

Comment:
can this and previous if be joined?

Reply:
Joined how?

Reply:
ah nvm, I thought the condition is identical